src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1982,6 +2149,53 @@ def translate_device(dev):
         else:
             return False
 
+    @private
+    def get_iommu_type(self):
+        def check_iommu(iommu_type):
+            IOMMU_TEST = {'VT-d': {'arglist': ['/usr/sbin/acpidump', '-t'],
+                                   'string': 'DMAR'},
+                          'amdvi': {'arglist': ['/sbin/sysctl', 'hw'],
```

Comment:
We could simply use `sysctl hw.vmm.amdvi.enable` ?
Btw we have a python module `sysctl` which can be used like this
```
sysctl.filter('hw.vmm.amdvi.enable')[0].value
```

Reply:
Yes, `sysctl hw.vmm.amdvi.enable` would certainly reduce the amount of output to parse. The `sysctl` python module looks nice too. Only drawback would seem to be that the detection mechanisms become further differentiated. I have made an attempt though to abstract the difference by including a detection mechanism/function in the IOMMU_TESTS dictionary. This way the different tests can still be looped over without conditions or exceptions. Maybe there is a better way?

Reply:
I think I can only advise once I see it ; )